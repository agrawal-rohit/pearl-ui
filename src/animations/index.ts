import { useAnimationState } from "../hooks/useAnimationState";

/**
 * Custom hook for creating a fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from an opacity of 0 to an opacity of 1.
 * This creates a fade-in effect.
 *
 * @returns An animation state that creates a fade-in effect.
 */
export const useFade = () => {
  // Define the initial and final states for the fade animation.
  // The animation will start with an opacity of 0 (completely transparent) and end with an opacity of 1 (completely opaque).
  const fadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
    },
    to: {
      opacity: 1,
    },
  });

  // Return the fade animation state.
  return fadeAnimationState;
};

/**
 * Custom hook for creating a scale and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scale of 0.5 and opacity of 0 to a scale of 1 and opacity of 1.
 * This creates a combined effect of scaling and fading in.
 *
 * @returns An animation state that creates a scale and fade-in effect.
 */
export const useScaleFade = () => {
  // Define the initial and final states for the scale and fade animation.
  // The animation will start with a scale of 0.5 (half the original size) and opacity of 0 (completely transparent)
  // and end with a scale of 1 (original size) and opacity of 1 (completely opaque).
  const scaleFadeAnimationState = useAnimationState({
    from: {
      scale: 0.9,
      opacity: 0,
    },
    to: {
      scale: 1,
      opacity: 1,
    },
  });

  // Return the scale and fade animation state.
  return scaleFadeAnimationState;
};

/**
 * Custom hook for creating a slide animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 to a translateX of 0.
 * This creates a slide-in effect.
 *
 * @returns An animation state that creates a slide-in effect.
 */
export const useSlide = () => {
  // Define the initial and final states for the slide animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and end with a translateX of 0 (original position).
  const slideAnimationState = useAnimationState({
    from: {
      translateY: -100,
    },
    to: {
      translateY: 0,
    },
  });

  // Return the slide animation state.
  return slideAnimationState;
};

/**
 * Custom hook for creating a slide and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 and opacity of 0 to a translateX of 0 and opacity of 1.
 * This creates a combined effect of sliding and fading in.
 *
 * @returns An animation state that creates a slide and fade-in effect.
 */
export const useSlideFade = () => {
  // Define the initial and final states for the slide and fade animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and opacity of 0 (completely transparent)
  // and end with a translateX of 0 (original position) and opacity of 1 (completely opaque).
  const slideFadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
      translateY: 10,
    },
    to: {
      opacity: 1,
      translateY: 0,
    },
  });

  // Return the slide and fade animation state.
  return slideFadeAnimationState;
};

/**
 * Custom hook for creating a collapse animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scaleY of 1 to a scaleY of 0.
 * This creates a collapse effect.
 *
 * @returns An animation state that creates a collapse effect.
 */
export const useCollapse = () => {
  // Define the initial and final states for the collapse animation.
  // The animation will start with a scaleY of 1 (original size) and end with a scaleY of 0 (collapsed).
  const collapseAnimationState = useAnimationState({
    from: {
      opacity: 0,
      height: 0,
    },
    to: {
      opacity: 1,
      height: 100,
    },
  });

  // Return the collapse animation state.
  return collapseAnimationState;
};
