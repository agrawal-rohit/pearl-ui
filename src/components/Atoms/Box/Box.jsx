import React from "react";
import { View } from "react-native";
import useStyledProps from "../../../hooks/useStyledProps";
import { useTheme } from "../../../hooks/useTheme";
import {
  backgroundColor,
  border,
  layout,
  opacity,
  position,
  shadow,
  spacing,
  visible,
} from "../../../theme/src/styleFunctions";
import {
  IBasePearlTheme,
  RestyleFunctionContainer,
} from "../../../theme/src/types";

// export type BoxProps<
//   Theme extends IBasePearlTheme
// > = BackgroundColorProps<Theme> &
//   OpacityProps &
//   VisibleProps &
//   LayoutProps &
//   SpacingProps<Theme> &
//   BorderProps<Theme> &
//   ShadowProps<Theme> &
//   PositionProps<Theme>;

export const boxRestyleFunctions = [
  backgroundColor,
  opacity,
  visible,
  layout,
  spacing,
  border,
  shadow,
  position,
];

/**
 * Box is the most abstract component on top of which all other Pearl UI components are built. By default, it renders a <View> element.
 */
// type ViewProps = React.ComponentProps<typeof View> & {
//   children?: React.ReactNode;
// };
// type ComponentProps<Theme extends IBasePearlTheme> = BoxProps<Theme> &
//   Omit<ViewProps, keyof BoxProps<Theme>>;

const Box = React.forwardRef((props, ref) => {
  const passedProps = useStyledProps(boxRestyleFunctions, props);

  return <View ref={ref} {...passedProps} />;
});

export default Box;
