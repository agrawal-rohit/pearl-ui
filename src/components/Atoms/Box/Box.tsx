import React from "react";
import { View, ViewProps } from "react-native";
import { pearlify } from "../../../pearlify";

/**
 * Box is the most abstract component on top of which all other Pearl UI components are built.
 * By default, it renders a <View> element.
 */
const Box = pearlify<ViewProps>(View, {
  componentName: "Box",
  type: "basic",
  animatable: true,
});

// The props that the Box component accepts.
export type BoxProps = React.ComponentProps<typeof Box>;

// Export the Box component as the default export.
export default Box;
