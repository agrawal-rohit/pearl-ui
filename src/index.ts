export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
