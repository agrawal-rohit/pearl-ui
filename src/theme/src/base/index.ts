import { breakpoints } from "./breakpoints";
import { BasePearlTheme, FinalPearlTheme } from "../types";

import { borderRadii } from "./borderRadii";
import { elevation } from "./elevation";
import { palette } from "./palette";
import { spacing } from "./spacing";
import { zIndices } from "./zIndices";
import {
  fontConfig,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
} from "./typography";

// Component configs
import SpinnerConfig from "../../../components/Atoms/Spinner/Spinner.config";
import ScreenConfig from "../../../components/Atoms/Screen/Screen.config";
import TextConfig from "../../../components/Atoms/Text/Text.config";
import ButtonConfig from "../../../components/Molecules/Button/Button.config";
import IconConfig from "../../../components/Atoms/Icon/Icon.config";
import TextLinkConfig from "../../../components/Molecules/TextLink/TextLink.config";
import InputConfig from "../../../components/Molecules/Input/Input.config";
import DividerConfig from "../../../components/Atoms/Divider/Divider.config";
import CheckBoxConfig from "../../../components/Molecules/CheckBox/CheckBox.config";
import ImageConfig from "../../../components/Molecules/Image/Image.config";
import RadioConfig from "../../../components/Molecules/Radio/Radio.config";
import BadgeConfig from "../../../components/Molecules/Badge/Badge.config";
import AvatarConfig from "../../../components/Molecules/Avatar/Avatar.config";
import { isFunction, mergeWith } from "lodash";

export const baseTheme = {
  palette,
  spacing,
  borderRadii,
  breakpoints,
  elevation,
  zIndices,
  fonts,
  fontConfig,
  fontSizes,
  fontWeights,
  lineHeights,
  components: {
    Icon: IconConfig,
    Text: TextConfig,
    Screen: ScreenConfig,
    Spinner: SpinnerConfig,
    Button: ButtonConfig,
    TextLink: TextLinkConfig,
    Input: InputConfig,
    CheckBox: CheckBoxConfig,
    Radio: RadioConfig,
    Badge: BadgeConfig,
    Divider: DividerConfig,
    Image: ImageConfig,
    Avatar: AvatarConfig,
  },
};

/**
 * Override particular parts of the default theme to create a custom theme as per you app's needs
 * @param overrideConfig Configuration overrides to be combined with the default theme
 * @returns
 */
export const extendTheme = <
  T extends typeof baseTheme | (Record<string, any> & {})
>(
  overrideConfig: T
) => {
  function customizer(source: any, override: any) {
    if (isFunction(source)) {
      return (...args: any[]) => {
        const sourceValue = source(...args);
        const overrideValue = isFunction(override)
          ? override(...args)
          : override;
        return mergeWith({}, sourceValue, overrideValue, customizer);
      };
    }
    return undefined;
  }

  const finalTheme = [overrideConfig].reduce((prevValue, currentValue) => {
    return mergeWith({}, prevValue, currentValue, customizer);
  }, baseTheme);

  return finalTheme as T & typeof baseTheme;
};
