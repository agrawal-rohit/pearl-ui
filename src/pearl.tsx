import {
  AtomComponent,
  ComponentTypes,
  FinalPearlTheme,
  MoleculeComponent,
  PearlComponent,
  StyleFunctionContainer,
} from "./theme/src/types";
import React from "react";
import { useAtomicComponentConfig } from "./hooks/useAtomicComponentConfig";
import { boxStyleFunctions, BoxStyleProps } from "./theme/src/style-functions";
import { useStyleProps } from "./hooks/useStyleProps";
import { useMolecularComponentConfig } from "./hooks/useMolecularComponentConfig";
import { MotiView, motify } from "moti";
import { useMotiWithStyleProps } from "./hooks/useMotiWithStyleProps";
import { useTheme } from "./hooks/useTheme";
import { Platform } from "react-native";

/**
 * Configuration object that adds metadata to a component that's required by Pearl UI
 */
interface pearlConfig {
  /**
   * The name of the component in the FinalPearlTheme
   */
  componentName: keyof FinalPearlTheme["components"];
  /**
   * The type of the component
   */
  type: ComponentTypes;
  /**
   * Whether the component is animatable or not
   */
  animatable?: boolean;
}

/**
 * Hook to convert a third-party component to a Pearl component that can be configured using style props, an atomic component configuration or a molecular component configuration.
 * @param Component The base component on which the pearlification has to be applied
 * @param config A configuration object that adds metadata to a component that's required by Pearl UI
 * @param styleFunctions The set of style functions that would define the style props that should be supported by the pearlified component
 * @param moleculeConfigOptions The options for configuring the molecular component
 * @returns
 */
export function pearl<
  ComponentProps,
  ComponentType extends ComponentTypes = "basic",
  ComponentAtoms extends Record<string, any> = Record<string, any>,
  StyleProps = BoxStyleProps,
  Animateable extends boolean = true,
>(
  Component: any,
  config: pearlConfig = {
    componentName: "None",
    type: "basic",
    animatable: true,
  },
  styleFunctions: StyleFunctionContainer[] = boxStyleFunctions,
  moleculeConfigOptions: {
    partForOverridenStyleProps?: string | undefined;
    partForOverridenNativeProps?: string | undefined;
    partForOverridenAnimationProps?: string | undefined;
  } = {}
) {
  let FinalComponent: any | undefined;
  const isAnimatable = !!config.animatable && config.type !== "molecule";

  if (isAnimatable) {
    const ConvertedClassComponent = class extends React.Component<any, any> {
      static displayName = `Pearl${
        Component.name ?? Component.displayName ?? `NoName`
      }`;

      render() {
        const { children, ...props } = this.props;
        return (
          <Component ref={(props as any).forwardedRef} {...props}>
            {children}
          </Component>
        );
      }
    };

    if (Platform.OS === "web" && config.componentName === "Box")
      FinalComponent = MotiView;
    else FinalComponent = motify(ConvertedClassComponent)();
  } else {
    FinalComponent = Component;
  }

  return React.memo(
    React.forwardRef(
      (
        {
          children,
          ...rest
        }: PearlComponent<ComponentProps, StyleProps, Animateable> &
          AtomComponent<(typeof config)["componentName"], ComponentType> &
          MoleculeComponent<(typeof config)["componentName"], ComponentType> & {
            atoms?: Partial<ComponentAtoms>;
            children?: string | JSX.Element | JSX.Element[] | React.ReactNode;
          },
        ref: any
      ) => {
        const { colorMode } = useTheme();

        let convertedProps;
        if (config.type === "atom") {
          convertedProps = useAtomicComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions
          );
          if (config.animatable)
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
        } else if (config.type === "molecule") {
          convertedProps = useMolecularComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions,
            moleculeConfigOptions.partForOverridenStyleProps,
            moleculeConfigOptions.partForOverridenNativeProps,
            moleculeConfigOptions.partForOverridenAnimationProps
          );
        } else {
          convertedProps = useStyleProps(rest, styleFunctions);
          if (config.animatable) {
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
          }
        }

        return (
          <FinalComponent
            {...convertedProps}
            key={
              Platform.OS === "web" &&
              ["Text", "Icon", "Box"].includes(config.componentName as string)
                ? (convertedProps as any).key!
                  ? `${(convertedProps as any).key!}-${colorMode}`
                  : `${colorMode}`
                : undefined
            }
            ref={ref}
          >
            {children}
          </FinalComponent>
        );
      }
    )
  );
}
