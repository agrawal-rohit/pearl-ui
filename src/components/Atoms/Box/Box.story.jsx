import React from "react";
import { storiesOf } from "@storybook/react-native";
import Screen from "../Screen/Screen";
import Box from "./Box";

storiesOf("Box", module)
  .addDecorator((getStory) => getStory())
  .add("Basic box", () => (
    <Box
      padding="s"
      height={300}
      width={100}
      backgroundColor="primary100"
    ></Box>
  ));
