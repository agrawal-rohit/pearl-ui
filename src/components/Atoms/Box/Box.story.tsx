import React from "react";
import { storiesOf } from "@storybook/react-native";
import Screen from "../Screen/Screen";
import Box from "../Box/Box";

storiesOf("Box", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("Basic box", () => (
    <Box padding="s" width={200} backgroundColor="primary100"></Box>
  ));
