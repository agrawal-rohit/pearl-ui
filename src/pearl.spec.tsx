import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "./theme/src/theme-context";
import { pearl } from "./pearl";
import { Text, View } from "react-native";
import { extendTheme } from "./theme/src/base/index";
import Box, { BoxProps } from "./components/atoms/box/box";

jest.useFakeTimers();

describe("pearl/basic", () => {
  const PearlViewComponent: React.FC<PropsWithChildren> = ({
    children,
    ...props
  }) => {
    const PearlView = pearl(View, {
      componentName: "TestView",
      type: "basic",
    });

    return <PearlView {...props}>{children}</PearlView>;
  };

  const PearlTextComponent: React.FC<PropsWithChildren> = ({
    children,
    ...props
  }) => {
    const PearlText = pearl(Text, {
      componentName: "TestText",
      type: "basic",
    });

    return <PearlText {...props}>{children}</PearlText>;
  };

  it("passes the snapshot test for a <View> component", () => {
    const tree = render(
      <ThemeProvider>
        <PearlViewComponent></PearlViewComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for a <Text> component", () => {
    const tree = render(
      <ThemeProvider>
        <PearlTextComponent>
          <Text>asdasd</Text>
        </PearlTextComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the correct children for a <View> component", () => {
    const { getByText } = render(
      <ThemeProvider>
        <PearlViewComponent>
          <Text>Test</Text>
        </PearlViewComponent>
      </ThemeProvider>
    );
    expect(getByText("Test")).toBeTruthy();
  });

  it("renders the correct children for a <Text> component", () => {
    const { getByText } = render(
      <ThemeProvider>
        <PearlTextComponent>
          <Text>Test</Text>
        </PearlTextComponent>
      </ThemeProvider>
    );
    expect(getByText("Test")).toBeTruthy();
  });
});

describe("pearl/atoms", () => {
  const newTheme = extendTheme({
    components: {
      TestView: {
        baseStyle: {
          width: 200,
          height: 200,
          backgroundColor: "danger.400",
          boxShadow: "4xl",
        },
      },
      TestText: {
        baseStyle: {
          color: "success.400",
          fontFamily: "body",
          fontWeight: "hairline",
        },
      },
    },
  });

  const PearlViewComponent: React.FC<PropsWithChildren> = ({
    children,
    ...props
  }) => {
    const PearlView = pearl(View, {
      componentName: "TestView",
      type: "atom",
    });

    return <PearlView {...props}>{children}</PearlView>;
  };

  const PearlTextComponent: React.FC<PropsWithChildren> = ({
    children,
    ...props
  }) => {
    const PearlText = pearl(Text, {
      componentName: "TestText",
      type: "atom",
    });

    return <PearlText {...props}>{children}</PearlText>;
  };

  it("passes the snapshot test for a <View> component", () => {
    const tree = render(
      <ThemeProvider theme={newTheme}>
        <PearlViewComponent></PearlViewComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for a <Text> component", () => {
    const tree = render(
      <ThemeProvider theme={newTheme}>
        <PearlTextComponent>asdasd</PearlTextComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the correct children for a <View> component", () => {
    const { getByText } = render(
      <ThemeProvider theme={newTheme}>
        <PearlViewComponent>
          <Text>Test</Text>
        </PearlViewComponent>
      </ThemeProvider>
    );
    expect(getByText("Test")).toBeTruthy();
  });

  it("renders the correct children for a <Text> component", () => {
    const { getByText } = render(
      <ThemeProvider theme={newTheme}>
        <PearlTextComponent>
          <Text>Test</Text>
        </PearlTextComponent>
      </ThemeProvider>
    );
    expect(getByText("Test")).toBeTruthy();
  });
});

describe("pearl/molecules", () => {
  const newTheme = extendTheme({
    components: {
      TestView: {
        parts: ["root", "box1", "box2"],
        baseStyle: {
          root: {
            width: 200,
            height: 200,
            backgroundColor: "danger.400",
            boxShadow: "4xl",
            p: "4xl",
          },
          box1: {
            width: 50,
            height: 50,
            backgroundColor: "success.500",
          },
        },
      },
    },
  });

  const PearlViewComponent: React.FC<BoxProps> = ({ children, ...props }) => {
    const MoleculeRender = (props: any) => {
      return (
        <Box {...props.root}>
          <Box {...props.box1}></Box>
          <Box {...props.box2}></Box>
        </Box>
      );
    };

    const PearlView = pearl(MoleculeRender, {
      componentName: "TestView",
      type: "molecule",
    });

    return <PearlView {...props}>{children}</PearlView>;
  };

  it("passes the snapshot test for a <View> component", () => {
    const tree = render(
      <ThemeProvider theme={newTheme}>
        <PearlViewComponent></PearlViewComponent>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
