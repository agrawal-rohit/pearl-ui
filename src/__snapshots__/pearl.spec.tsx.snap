// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pearl/atoms passes the snapshot test for a <Text> component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    color="success.400"
    fontFamily="body"
    fontWeight="hairline"
    style={{}}
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    asdasd
  </Text>
</View>
`;

exports[`pearl/atoms passes the snapshot test for a <View> component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    box1={
      {
        "backgroundColor": "success.500",
        "height": 50,
        "width": 50,
      }
    }
    root={
      {
        "backgroundColor": "danger.400",
        "boxShadow": "4xl",
        "height": 200,
        "p": "4xl",
        "width": 200,
      }
    }
    style={
      {
        "backgroundColor": "#FF8371",
        "elevation": 28,
        "height": 200,
        "shadowColor": "#1A2138",
        "shadowOffset": {
          "height": 28,
          "width": 0,
        },
        "shadowOpacity": 0.16,
        "shadowRadius": 28,
        "width": 200,
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  />
</View>
`;

exports[`pearl/basic passes the snapshot test for a <Text> component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={{}}
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    asdasd
  </Text>
</View>
`;

exports[`pearl/basic passes the snapshot test for a <View> component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={{}}
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  />
</View>
`;

exports[`pearl/molecules passes the snapshot test for a <View> component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={{}}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={{}}
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <View
      animate={{}}
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={{}}
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
    <View
      animate={{}}
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={{}}
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
</View>
`;
